api/producer/writer.go

```diff
@@ -0,0 +1,105 @@
+/*
+   Copyright © 2024 The CDI Authors
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package producer
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+
+	cdi "tags.cncf.io/container-device-interface/specs-go"
+)
+
+// A SpecWriter defines a structure for outputting CDI specifications.
+type SpecWriter struct {
```

Comment:
Can this struct be non-public. It has only non-public fields.

Reply:
The issue is that we return this type from our `NewSpecWriter` constructure. We could return an interface instead, but I don't think that that's a requirement.

Reply:
even if we return non-public struct it's still possible to call public methods on it IIRC.

Reply:
OK, makes sense.